### ISSUE TITLE
Express matching fails with absolute urls

### ISSUE THREAD

**Issue by kirkov** [wheresrhys/fetch-mock#371]
Given this express path match: `express:/apps/:id` it generates the following regex: `^\/apps\/([^\/]+?)(?:\/)?$`.
However the url it tests against is absolute: `http://example.com/apps/xxx`.
So the regex doesn't match. 

I can't find a way around this. The Path-to-regex library supports disabling the "start" (^) matching, (https://github.com/pillarjs/path-to-regexp#usage) and that would solve the issue as it then only matches with ends with ($).

Not sure of the impact though.

Here's a test to reproduce:
```js
/* eslint-env jest */
const fetchMock = require('fetch-mock')

it('express match full url', () => {
  fetchMock.mock('express:/apps/:id', 200)

  fetch('https://api.example.com/apps/xxx')
    .then(res => {
      expect(res.ok).toBe(true)
    })
})
```

**Comment by kirkov** [comment]
Oh, just found out that the path-to-regexp library supports "match all" by using `(.*)`, so this will match a full url: `express:(.*)/apps/:id`. 

Maybe just a docs update is enough then.

**Comment by wheresrhys** [comment]
Thanks for reporting this. I have a PR to fix it https://github.com/wheresrhys/fetch-mock/compare/express-absolute?expand=1. Gonna release as a patch as it has the feel of a bug.

**Comment by wheresrhys** [comment]
fixed in 7.0.6

**Comment by kirkov** [comment]
@wheresrhys Thanks for the quick fix! 🙌
